libs/react/src/lib/segmented-control/segmented-control.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { ReactNode } from 'react'
+import { createComponent } from '@lit/react'
+import { GdsSegmentedControl } from '@sebgroup/green-core/components/segmented-control/index.js'
+import { GdsSegment } from '@sebgroup/green-core/components/segmented-control/segment/index.js'
+import { getScopedTagName } from '@sebgroup/green-core/scoping'
+import { registerTransitionalStyles } from '@sebgroup/green-core/transitional-styles'
+import classNames from 'classnames'
+
+registerTransitionalStyles()
+
+export const CoreSegementedControl = createComponent({
+  tagName: getScopedTagName('gds-segmented-control'),
+  elementClass: GdsSegmentedControl,
+  events: { onchange: 'change' },
+  react: React,
+})
+
+export const CoreSegment = createComponent({
+  tagName: getScopedTagName('gds-segment'),
+  elementClass: GdsSegment,
+  react: React,
+})
+
+export interface SegmentedControlProps {
+  value?: string
+  segMinWidth?: number
+  size?: 'small' | 'medium'
+  onChange?: (value: string) => void
+  className?: string
+  children?: ReactNode | ReactNode[]
+}
+
+export interface SegmentProps {
+  value?: string
+  className?: string
+  disabled?: boolean
+  children?: ReactNode | ReactNode[]
+}
+
+const SegmentedControl: React.FC<SegmentedControlProps> = ({
+  onChange,
+  className,
+  ...props
+}: SegmentedControlProps) => {
+  const onControlChange = (event: Event) => {
```

Comment:
For consistency, I think `onChange` should just be forwarded directly here, and let the consumer deal with the event. That's how it works in the other wrappers, unless backwards-compatiblity reasons prevent it, and it's also more in line with how events work in native components, like `<input>` and `<button>`.